lib/experimental/html/class-wp-html-tag-processor.php

```diff
@@ -397,6 +397,10 @@ public function next_tag( $query = null ) {
 		$already_found = 0;
 
 		do {
+			if ( $this->parsed_bytes >= strlen( $this->html ) ) {
```

Comment:
Sorry for the drive by review but I just noticed that `strlen( $this->html )` is being called throughout this class. Would this be optimized by storing the length in a `$html_length` member variable in the constructor and the `get_updated_html` method, and then using it instead?

Reply:
thanks for the question @westonruter 

the answer I believe is _no_ because `strlen` isn't like a normal PHP function call, it's [a PHP op-code](https://github.com/php/php-src/blob/master/Zend/zend_builtin_functions.c#L291) and the execution is [a direct lookup on the internal string object](https://github.com/php/php-src/blob/master/Zend/zend_string.h#L67) within the PHP runtime. maybe it saves one indirect memory access, but if that's true I'm not sure it'd be measurable (whereas a function call is often measurable!)

I haven't confirmed this but the docs inside PHP suggest this might even be special-cased and there's truly no difference in the generated CPU instructions. the opcode output is different, but not in a way that sheds light on the CPU instructions (other than that it's not a function call with their characteristic overhead).

so we have a lint rule floating around I think that discourages this style but I believe it's based on a false premise.